### PR TITLE
Add support for date/time datas on plot 2d

### PR DESCRIPTION
**Pull request by Dalan94**
This is my first implementation of date/time formatting for axes (see #9).

Please feel free to ask me some modifications.

I'm particullary not proud of the plotdata implementation.
I add an DateTime vector because with the plplot API, I need to have a plstream to convert DateTime to double. But it duplicate X datas so it's bad.

I thought about create new class but I will lose all Plot2D attribute.

### CODE REVIEW

**Comment by tschoonj** [comment]
While I greatly appreciate the work you put into this, I think this could be simplified quite a bit if you don't modify PlotData2D at all, and instead focus solely on Plot2D. Essentially if the user wants to treat the data as time, then he should use a setter to specify the time format. Only if different from the empty string, the data will be interpreted as timestamps according to the plplot convention which assumes that the values will be seconds since the 1/1/1970 epoch.


You are right to include a check for logarithmic plotting though: I hadnt considered that...

File: gtkmm-plplot/plot2d.cpp [hunk 1]
```diff
@@ -184,6 +187,8 @@ void Plot2D::set_axis_logarithmic_x(bool _log10) {
 void Plot2D::set_axis_logarithmic_y(bool _log10) {
   //need to check that all values are positive!
   if (_log10) {
+    if (get_axis_logarithmic_x())
```
Comment (tschoonj) [inline gtkmm-plplot/plot2d.cpp]: I don't understand why you have modified `Plot2D::set_axis_logarithmic_y` at all if you're only supporting datetimes on the X-axis (for now?)

Perhaps something like this should go into `Plot2D::set_axis_logarithmic_x`? But even then you would need to check if datetimes are going to be used before throwing an exception.

File: gtkmm-plplot/plot2d.cpp [hunk 2]
```diff
@@ -204,6 +209,28 @@ bool Plot2D::get_axis_logarithmic_y() {
   return log10_y;
 }
 
+void Plot2D::set_axis_time_format(Glib::ustring _time_format)
+{
+  if (!_time_format.empty())
+  {
+    if (get_axis_logarithmic_x())
+      throw Exception("Gtkmm::Plplot::Plot2D::set_axis_time_format -> X-axys cannot be time and logarithmic");
```
Comment (tschoonj) [inline gtkmm-plplot/plot2d.cpp]: please replace axys with axis.

Comment (tschoonj) [reply gtkmm-plplot/plot2d.cpp]: also please follow my convention for positioning curly braces

Comment (tschoonj) [reply gtkmm-plplot/plot2d.cpp]: Are you not planning to support datetimes on the Y-axis as well? You should rename `Plot2D::set_axis_time_format` to `Plot2D::set_axis_time_format_x` and `Plot2D::get_axis_time_format` to `Plot2D::get_axis_time_format_x`.

File: gtkmm-plplot/plot2d.cpp [hunk 3]
```diff
@@ -271,6 +298,17 @@ void Plot2D::draw_plot(const Cairo::RefPtr<Cairo::Context> &cr, const int width,
       plplot_axis_style += PLPLOT_LIN_LIN;
   }
 
+  // Set time
+  if (!time_format.empty())
+  {
+    pls->timefmt(time_format.c_str());
+    plplot_axis_style += PLPLOT_DATE_LIN;
+
+	pls->configtime(time_scale, 0,0,0,1,time_start.get_year(), time_start.get_month() - 1,
```
Comment (tschoonj) [inline gtkmm-plplot/plot2d.cpp]: Please fix the indentation here

**Comment by tschoonj** [comment]
Many thanks for updating the code. I will play a bit with it and if I have no further comments merge it in.

**Comment by tschoonj** [comment]
Ok, looks good. Any plans for adding Y-axis support as well?

**Comment by Dalan94** [comment]
No, I don't need it and I'm not sure that's very useful.
If someone wants to, he can implement it without breaking the API.

**Comment by tschoonj** [comment]
Ok thanks again!

**Pull request merged**